src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let newState = Object.assign({}, state);
+  let lastCopy = 0;
+
+  for (const action of actions) {
+    switch (action.type) {
```

Comment:
```suggestion
    const previousState = lastStateIndex >= 0
      ? transformedStates[lastStateIndex]
      : state;
    const currentState = Object.assign({}, previousState);
    
    switch (action.type) {
```

Use a copy of the previous state on every iteration